framework/src/main/java/org/checkerframework/common/aliasing/AliasingTransfer.java

```diff
@@ -83,41 +87,81 @@ public TransferResult<CFValue, CFStore> visitAssignment(
+        } else if (n instanceof ObjectCreationNode) {
+            objectCreationNode = (ObjectCreationNode) n;
+        } else {
+            throw new BugInCF(
+                    "processPostconditions in AliasingTransfer expects "
+                            + "a MethodInvocationNode or ObjectCreationNode argument; received a "
+                            + n.getClass().getSimpleName());
+        }
         super.processPostconditions(n, store, methodElement, tree);
-        if (TreeUtils.isEnumSuper(n.getTree())) {
+        if (methodInvocationNode != null && TreeUtils.isEnumSuper(methodInvocationNode.getTree())) {
             // Skipping the init() method for enums.
             return;
         }
-        List<Node> args = n.getArguments();
+        List<Node> args =
+                methodInvocationNode != null
+                        ? methodInvocationNode.getArguments()
+                        : objectCreationNode.getArguments();
         List<? extends VariableElement> params = methodElement.getParameters();
-        assert (args.size() == params.size())
-                : "Number of arguments in "
-                        + "the method call "
-                        + n
-                        + " is different from the"
-                        + " number of parameters for the method declaration: "
-                        + methodElement.getSimpleName();
-
+        // Only check MethodInvocationNode
+        if (methodInvocationNode != null) {
+            assert (args.size() == params.size())
+                    : "Number of arguments in "
+                            + "the method call "
+                            + n
+                            + " is different from the"
+                            + " number of parameters for the method declaration: "
+                            + methodElement.getSimpleName();
+        }
+        // use adaptParameters for object creation node
+        if (objectCreationNode != null) {
+            AnnotatedTypeFactory.ParameterizedExecutableType fromUse =
+                    factory.constructorFromUse((NewClassTree) tree);
+            AnnotatedExecutableType invokedMethod = fromUse.executableType;
+            List<AnnotatedTypeMirror> objectParams =
+                    AnnotatedTypes.adaptParameters(
+                            factory, invokedMethod, ((NewClassTree) tree).getArguments());
+            List<?> passedArgs = ((NewClassTree) tree).getArguments();
+            assert (passedArgs.size() == objectParams.size())
+                    : "Number of arguments in "
+                            + "the object creation "
+                            + n
+                            + " is different from the"
+                            + " number of parameters for the method declaration: "
+                            + methodElement.getSimpleName();
+        }
         AnnotatedExecutableType annotatedType = factory.getAnnotatedType(methodElement);
```

Comment:
For an object creation, on line 128 you already determined a type. What is the relation? Should this be the same?

Reply:
Thanks! Let me try...